notes-0225-universes.tex

```diff
@@ -0,0 +1,272 @@
+\documentclass[11pt]{article}
+\usepackage{hdtt2020}
+\usepackage{newpxmath}
+\usepackage{newpxtext}
+
+\newcommand{\U}{\ensuremath{\mathcal{U}}}
+
+\newcommand{\wf}[1]{\ensuremath{\textnormal{well-founded}(#1)}}
+\newcommand{\isequiv}[1]{\ensuremath{\textnormal{is-equiv}(#1)}}
+\newcommand{\qinv}[1]{\ensuremath{\textnormal{qinv}(#1)}}
+\newcommand{\isprop}[1]{\ensuremath{\textnormal{is-prop}(#1)}}
+\newcommand{\iscontr}[1]{\ensuremath{\textnormal{is-contr}(#1)}}
+\newcommand{\ishae}[1]{\ensuremath{\textnormal{is-hae}(#1)}}
+\newcommand{\ap}{\textnormal{ap}}
+\newcommand{\idtoequiv}{\textnormal{idtoequiv}}
+\newcommand{\funapp}{\textnormal{funapp}}
+
+\title{CSCI 8980 Higher-Dimensional Type Theory\\ Lecture Notes}
+\author{Dawn Michaelson, Jack Perisich}
+\date{February 25, 2020}
+
+\begin{document}
+
+\maketitle
+
+
+
+
+
+\section{Universe}
+To internalize the $\istype{\Gamma}{A}$ judgment, we introduce the
+universe type \U.  This can be viewed as the ``type of types''.  We
+replace the original judgment with $\oftype{\Gamma}{A}{\U}$.
+%
+We can make this replacement in all our existing rules.  For example,
+we originally had the following rule for dependent function types:
+\begin{prooftree*}
+  \hypo{\istype{\Gamma}{A}}
+  \hypo{\istype{\Gamma, x:A}{B}}
+  \infer2{\istype{\Gamma}{\dfuntype{x}{A}{B}}}
+\end{prooftree*}
+We can replace each judgment with a judgment that the given type has
+the type of the universe, giving us a new rule:
+\begin{prooftree*}
+  \hypo{\oftype{\Gamma}{A}{\U}}
+  \hypo{\oftype{\Gamma, x:A}{B}{\U}}
+  \infer2{\oftype{\Gamma}{\dfuntype{x}{A}{B}}{\U}}
+\end{prooftree*}
+Doing this in all rules gives a system where we do not need the
+original is-type judgment.
+
+
+
+
+
+\section{Inconsistency}
+Because \U{} is a type, we should have a rule:
+\begin{prooftree*}
+  \infer0{\oftype{\Gamma}{\U}{\U}}
+\end{prooftree*}
+With this rule, we can prove inconsistency, meaning a closed term of
+the empty type.  This is the \textbf{Burali-Forti Paradox}.
+\begin{enumerate}
+\item Consider types with a transitive, well-founded order.  We will
+  write this as \wf{A}.  This means that the type has an induction
+  principle and a minimal element.
```

Comment:
Clarification: It might not have a minimal element, but a minimal element _for any non-empty subset._ This condition is (classically at least) equivalent to having an induction principle. The empty type with an empty relation is well-founded according to our definition.